ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,1187 @@
+         *   console.log(button);
+         * }
+         * ```
+         *
+         * Return false to stop default buttons click action.
+         *
+         * @type      {Highcharts.BreadcrumbsClickCallbackFunction}
+         * @since     next
+         * @apioption breadcrumbs.events.click
+         */
+
+        /**
+         * When the breadcrumbs are floating, the plot area will not move
+         * to make space for it. By default, the chart will not make space
+         * for the buttons. This property won't work when positioned in the
+         * middle.
+         *
+         * @sample {highcharts} highcharts/breadcrumbs/floating-false
+         *          Non-floating breadcrumbs
+         *
+         * @type      {boolean}
+         * @since     next
+         */
+        floating: false,
+
+        /**
+         * A format string for the breadcrumbs button.
+         * Variables are enclosed by curly brackets.
+         * Available values are passed in the declared point options.
+         *
+         * @type      {string|undefined}
+         * @since     next
+         * @default   undefined
+         * @sample {highcharts} highcharts/breadcrumbs/format
+         *          Display custom values in breadcrumb button.
+         */
+        format: void 0 as any,
+
+        /**
+         * Callback function to format the breadcrumb text from scratch.
+         *
+         * @type      {Highcharts.BreadcrumbsFormatterCallbackFunction}
+         * @since     next
+         * @apioption breadcrumbs.formatter
+         */
+
+        /**
+         * What box to align the button to. Can be either `plotBox` or
+         * `spacingBox`.
+         *
+         * @type       {Highcharts.ButtonRelativeToValue}
+         * @default    plotBox
+         * @since      3.0.8
+         * @product    highcharts highmaps
+         * @apioption  drilldown.breadcrumbs.relativeTo
+         */
+        relativeTo: 'plotBox',
+
+        /**
+         * Positioning for the button row. The breadcrumbs buttons
+         * will be aligned properly for the default chart layout
+         * (title,  subtitle, legend, range selector) for the custom chart
+         * layout set the position properties.
+         * @sample {highcharts} highcharts/breadcrumbs/position
+         *          Custom button positioning.
+         */
+        position: {
+            /**
+             * Horizontal alignment of the breadcrumbs buttons.
+             *
+             * @type {Highcharts.AlignValue}
+             */
+            align: 'left',
+
+            /**
+             * Vertical alignment of the breadcrumbs buttons.
+             *
+             * @type {Highcharts.VerticalAlignValue}
+             */
+            verticalAlign: 'top',
+
+            /**
+             * The X offset of the breadcrumbs button group.
+             */
+            x: 0,
+
+            /**
+             * The Y offset of the breadcrumbs button group.
+             */
+            y: 0
```

Comment:
This doesn't match with the fact that we are reserving space for the button above the plot area. Probably the default `y` should be negative. https://jsfiddle.net/highcharts/fyqL74nj/

Reply:
Yes, actually, I had a question in this matter as well. 

Do we want to reserve space at all? I thought that when a group is relative to `plotBox` and similar, then we should not reserve space - buttons are inside plotArea so no extra margin is needed. 

On the other hand, I would like to reserve this space when group is relative to a `chart`. But then we need to move the button a little bit lower, so the default 0 position won't really be at the top of a chart anymore.

Reply:
I made a modification so that `y` is now automatically set based of the `floating` option.

Reply:
Ok, sounds great :)